### PR TITLE
Fixed: calling sync from an async GCD queue + subtle memory management issue

### PR DESCRIPTION
**Pull request by jordansmithnz**

### CODE REVIEW

File: Kronos.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -330,6 +346,7 @@
 				GCC_WARN_UNUSED_VARIABLE = YES;
 				IPHONEOS_DEPLOYMENT_TARGET = 8.0;
 				MTL_ENABLE_DEBUG_INFO = NO;
+				SWIFT_OPTIMIZATION_LEVEL = "-Owholemodule";
```
Comment (Reflejo) [inline Kronos.xcodeproj/project.pbxproj]: Can you revert project and scheme changes?

File: Tests/KronosTests/DNSResolverTests.swift [hunk 2]
```diff
@@ -52,4 +52,15 @@ final class DNSResolverTests: XCTestCase {
...
     }
+    
+    func testTemporaryRunloopHandling() {
+        let expectation = self.expectation(description: "Query works from async GCD queues")
+        DispatchQueue(label: "Ephemeral DNS test queue").async {
+            DNSResolver.resolve(host: "lyft.com") { _ in
+                expectation.fulfill()
+            }
+        }
+
```
Comment (Reflejo) [inline Tests/KronosTests/DNSResolverTests.swift]: Remove trailing spaces (see linter failure @ Travis)

**Comment by Reflejo** [comment]
Thanks for doing this, looking good. Just a few comments

**Comment by jordansmithnz** [comment]
No worries, fixes pushed now.

**Comment by Reflejo** [comment]
@jordansmithnz Awesome, can you squash? I'll merge after you do. Also curious as to what's the "memory management issue" you are referring to

**Comment by jordansmithnz** [comment]
Yeah, done. I thought I knew, until I had another look at it just before... One of the tests was sporadically failing in NTPClientTests.swift. Made it through line 13, and caused an unwrap error on line 15, which seemed a little off.

The change I made to 'fix' it was never intended as a fix, but the test stopped failing after that, pretty consistently. I put it down to some sort of weird swift compiler bug since it should have been equivalent.

Looks like that was all just chance though, because it failed with the same behaviour again just now.

**Comment by Reflejo** [comment]
@jordansmithnz Can you squash into 1 commit?. Also one more thing (sorry!) please sign [Contributor License Agreement (CLA)](https://oss.lyft.com/cla)

**Comment by jordansmithnz** [comment]
Ohh, yep, I see what you're saying. Will do.

**Comment by Reflejo** [comment]
Thanks a lot @jordansmithnz

**Pull request merged**